test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -3931,24 +3931,22 @@ def log_lines(self, logtype, id=None, n=50, tail=True, starttime=None,
                 dayend = time.strftime("%Y%m%d", time.localtime(endtime))
                 firstday_obj = datetime.datetime.strptime(daystart, '%Y%m%d')
                 lastday_obj = datetime.datetime.strptime(dayend, '%Y%m%d')
+                if logtype == 'accounting':
+                    logdir = os.path.join(self.pbs_conf['PBS_HOME'],
+                                          'server_priv', 'accounting')
+                    sudo = True
+                elif (isinstance(self, Scheduler) and
+                        'sched_log' in self.attributes):
+                    logdir = self.attributes['sched_log']
+                else:
+                    logval = self._instance_to_logpath(logtype)
+                    if logval is None:
+                        m = 'Invalid logtype'
+                        raise PtlLogMatchError(rv=False, rc=-1, msg=m)
+                    logdir = os.path.join(self.pbs_conf['PBS_HOME'], logval)
                 while firstday_obj <= lastday_obj:
                     day = firstday_obj.strftime("%Y%m%d")
-                    if logtype == 'accounting':
-                        filename = os.path.join(self.pbs_conf['PBS_HOME'],
-                                                'server_priv', 'accounting',
-                                                day)
-                        sudo = True
-                    else:
-                        if (isinstance(self, Scheduler) and
-                                'sched_log' in self.attributes):
-                            filename = os.path.join(
-                                self.attributes['sched_log'], day)
-                        else:
-                            logval = self._instance_to_logpath(logtype)
-                            if logval:
-                                filename = os.path.join(
-                                    self.pbs_conf['PBS_HOME'],
-                                    logval, day)
+                    filename = os.path.join(logdir, day)
```

Comment:
Sorry for not asking before, if logdir, filename or day_lines fails, will it be caught in the except block?

Reply:
Yeah it will be caught by except block.